### PR TITLE
15.1 docs: `forbidden`, `unauthorized`, and `authInterrupts`

### PR DESCRIPTION
**Pull request by delbaoliveira**
**Do not merge until https://github.com/vercel/next.js/pull/72785 lands in canary.**

Closes:
- https://linear.app/vercel/issue/DOC-3820/[unstable]-forbidden-and-forbiddenjs
- https://linear.app/vercel/issue/DOC-3849/[unstable]-unauthorized-and-unauthorizedjs
- https://linear.app/vercel/issue/DOC-3860/[experimental]-authinterrupts-config-option

### CODE REVIEW

File: docs/01-app/03-api-reference/04-functions/unstable_forbidden.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - app/api-reference/file-conventions/forbidden
+---
+
+The `unstable_forbidden` function allows you to render the [`forbidden.js` file](/docs/app/api-reference/file-conventions/forbidden) during authentication.
+
+## Version History
+
+| Version   | Changes                          |
+| --------- | -------------------------------- |
+| `v15.1.0` | `unstable_forbidden` introduced. |
```
Comment (huozhi) [inline docs/01-app/03-api-reference/04-functions/unstable_forbidden.mdx:16]: we can also remove the prefix here. The feature is guarded under the flag `experimental. authInterrupts` in next config

File: docs/01-app/03-api-reference/04-functions/unstable_unauthorized.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    - app/api-reference/file-conventions/unauthorized
+---
+
+The `unstable_unauthorized` function allows you to render the [`unauthorized.js` file](/docs/app/api-reference/file-conventions/unauthorized) during authentication.
+
+## Version History
+
+| Version   | Changes                             |
+| --------- | ----------------------------------- |
+| `v15.1.0` | `unstable_unauthorized` introduced. |
```
Comment (huozhi) [inline docs/01-app/03-api-reference/04-functions/unstable_unauthorized.mdx:16]: same, we don't have the `unstable_` prefix.

**Review by molebox — APPROVED** [review]

File: docs/01-app/03-api-reference/03-file-conventions/unauthorized.mdx [hunk 3]
```diff
@@ -0,0 +1,114 @@
+---
+title: unauthorized.js
+description: API reference for the unauthorized.js special file.
+related:
+  links:
+    - app/api-reference/functions/unauthorized
+version: experimental
```
Comment (delbaoliveira) [inline docs/01-app/03-api-reference/03-file-conventions/unauthorized.mdx:7]:
```suggestion
version: canary
```

File: docs/01-app/03-api-reference/05-config/01-next-config-js/authInterrupts.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+---
+title: authInterrupts
+description: Learn how to enable the experimental `authInterrupts` configuration option to use `forbidden` and `unauthorized`.
+version: experimental
```
Comment (delbaoliveira) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/authInterrupts.mdx:4]:
```suggestion
version: canary
```

**Pull request merged**